### ISSUE TITLE
Bridge elevations should interpolate elevations at the ends

### ISSUE THREAD

**Issue by graue** [bikehopper/graphhopper#164]
Crossing Geneva over 280, the router thinks you're descending and ascending extremely steep hills:

![Screenshot from 2024-11-24 18-42-47](https://github.com/user-attachments/assets/f4c7665f-4c70-4242-8e02-a330773c8349)

![Screenshot from 2024-11-24 18-43-11](https://github.com/user-attachments/assets/3eb1e66f-b5cf-4ad9-9f96-669a3a3500df)

Stretches of road with [`bridge=*`](https://wiki.openstreetmap.org/wiki/Key:bridge) should perhaps treat the elevation as linearly interpolated between the ends.

**Comment by graue** [comment]
Thanks, Peter! Though it apparently doesn't currently work in our fork, based on those grade numbers that we see, but it's helpful to have that pointer on what code we should look at.

**Comment by abhumbla** [comment]
it's very possible we broke something when we did all that elevation penalty work a while ago

**Comment by abhumbla** [comment]
that's probably it. we now store the calculated grade on the edge at graph creation time so that we can refer to it in the flag encoder step when assigning weights

we use both grade and infrastructure quality to determine the penalty multiplier for each edge. this means grade is double counted, since we also reduce the speed (and increase the time) for high grades. but this seems to be warranted as it's more accurately capturing the effort cost of steep blocks. the relevant PR is #101 . maybe there was a better way to do this, but this works

the router now generally understands hilly San Francisco (here it is correctly navigating [the Wiggle] around Buena Vista park

![Screenshot_20241205-161423](https://github.com/user-attachments/assets/b8db4ac6-dadc-4383-8a6b-365833e4379b)

the bus provides a nice lift with a more direct route 

![Screenshot_20241205-161542](https://github.com/user-attachments/assets/5f00c3a2-4136-4488-8e67-356c5542936d)

**Comment by abhumbla** [comment]
of course, you know this much better than us! so any suggestions are very welcome :)
